### PR TITLE
Move cvdmirror

### PR DESCRIPTION
**Pull request by skifahrer**
This PR introduces the final missing piece to the scanning infrastructure—**a mirror of CVD files**. The mirror is updated regularly to maintain the latest CVD (ClamAV Virus Database) and distribute it to `clamcv` instances.

#### Why Do We Need It?

The CVD distributor limits the number of connections per IP to twice every 24 hours. If you exceed this limit, you get blocked and are unable to run ClamAV.

#### Key Changes:

1. **CVD Mirror Integration**:
   - A mirror for CVD files is being added to ensure the newest virus database is available for distribution to `clamav` instances.
   - The mirror is updated regularly to maintain up-to-date virus signatures.

2. **Updated Pipeline Precedence**:
   The scanning pipeline precedence is now as follows:
   1. `cvdmirror`: Deployed first and holds the latest CVD files.
   2. `clamav`: Downloads the CVD database from `cvdmirror`.
   3. `nest-clamav-scanner`: Relies on `clamav` for scanning, requiring `clamav` to be running.

3. **Exclusion of `forms-shared`**:
   - `forms-shared` no longer needs to be built during scanner infrastructure deployment since it’s not required.
   - This change reduces build time and speeds up the deployment process.

### CODE REVIEW

**Comment by mpinter** [comment]
forwarding the review to @vidriduch

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -21,7 +21,6 @@ jobs:
 
   conditions:
     name: Check for cluster conditions
-    needs: build-forms-shared
```
Comment (skifahrer) [inline .github/workflows/deploy.yml:24]: now forms shared are not build on every run, but only on those instances where are needed

File: .github/workflows/cluster-deploy-conditions-konto.yml [hunk 2]
```diff
@@ -217,9 +245,17 @@ jobs:
...
+        if: startsWith(github.ref, 'refs/tags/prod-cvdmirror') || steps.prod.outputs.condition == 'true'
+        run: |
+          echo "COUNT=$(($COUNT+1))" >> $GITHUB_ENV
+          echo "condition=true" >> $GITHUB_OUTPUT
+          echo ":soon: **prod-cvdmirror**" >> $GITHUB_STEP_SUMMARY
+
       - id: prod-clamav
         name: Check for prod clamav
-        if: startsWith(github.ref, 'refs/tags/prod-clamav') || steps.prod.outputs.condition == 'true'
+        if: startsWith(github.ref, 'refs/tags/prod-clamav') || steps.staging.outputs.condition == 'true'
```
Comment (vidriduch) [inline .github/workflows/cluster-deploy-conditions-konto.yml:258]: Why has `prod-clamav` staging condition?

File: cvdmirror/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Copy config
+COPY Caddyfile Caddyfile
+
+# Copy Scripts
+COPY health.sh health.sh
+COPY entrypoint.sh entrypoint.sh
+
+# Set permissions
+RUN chmod +x ./health.sh
+RUN chmod +x ./entrypoint.sh
```
Comment (vidriduch) [inline cvdmirror/Dockerfile:21]: I'm guessing this can be executed in just one `RUN` instruction

**Review by vidriduch — CHANGES_REQUESTED** [review]

File: .github/workflows/cluster-deploy-conditions-konto.yml (hunk 2 above)
Comment (skifahrer) [reply .github/workflows/cluster-deploy-conditions-konto.yml]: No it shouldn't be there. It's an old typo that I fixed in ClamAV branch afterwards, but this branch was created before the fix, so the typo remains. It`s now fixed.

File: cvdmirror/Dockerfile (hunk 3 above)
Comment (skifahrer) [reply cvdmirror/Dockerfile]: improved

File: cvdmirror/health.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env /bin/sh
+
+if [ "$(echo PING | nc 127.0.0.1 8080)" = "PONG" ]; then
+	echo "ping successful"
+else
+	echo "ping failed"
+	exit 1
+fi
```
Comment (vidriduch) [inline cvdmirror/health.sh:8]: (optional) Missing end lines in bunch of files

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -87,9 +86,24 @@ jobs:
       service-account: ${{ secrets.DEV_STANDALONE_TOKEN }}
       registry-pass: ${{ secrets.HARBOR_REGISTRY_PASSWORD }}
 
+  deploy-dev-cvdmirror:
+    name: dev cvdmirror
+    needs: conditions
+    if: needs.conditions.outputs.dev-cvdmirror == 'true'
+    uses: bratislava/github-actions/.github/workflows/deploy-with-bratiska-cli-inhouse.yml@beta
```
Comment (vidriduch) [inline .github/workflows/deploy.yml:93]: I'm not totally sure if this should be `@beta` couse of DEV or it needs to be changed before release to `@stable`.

**Review by vidriduch — APPROVED** [review]

File: cvdmirror/health.sh (hunk 4 above)
Comment (skifahrer) [reply cvdmirror/health.sh]: fixed

File: .github/workflows/deploy.yml (hunk 5 above)
Comment (skifahrer) [reply .github/workflows/deploy.yml:93]: I usually leave `dev` deployments with the `beta` tag, which I use for testing future updates on pipelines. It’s good to keep it distinct from `stable`.

**Pull request merged**